### ISSUE TITLE
Application panics during runtime when compiled with Apple clang 14

### ISSUE THREAD

**Issue by MrMuetze** [tokio-rs/tokio#5287]
**Version**
```
tokio v1.22.0
tokio-macros v1.8.0 (proc-macro)
tokio-stream v0.1.11
tokio-util v0.7.4
tokio-io-timeout v1.2.0
tokio-rustls v0.23.4
```

**Platform**
`Darwin Mac-mini 21.6.0 Darwin Kernel Version 21.6.0: Thu Sep 29 20:12:57 PDT 2022; root:xnu-8020.240.7~1/RELEASE_X86_64 x86_64`
Rust toolchain: `nightly-2022-11-04`

**Description**
Hi everyone! At my company we have a large project that uses Tokio as part of our network layer. Recently a small code change in one of our protobuf messages has caused the application to fail with a runtime crash (it is a fairly simpel code change that doesn't touch anything low level). The backtrace points to a panic in `block.rs:363`. I will paste the backtrace further below.

I am the only one using macOS at my company and interestingly, I am also the only one with this issue. Other setups using Linux, Windows or other Unixes show no problems. I was able to reproduce this issue on two of my macOS machines, my office Mac mini and my "work from home" MacBook Pro (2016). Both use Apple clang 14 to compile a C++ application that links our Rust library (`libgst_api-3`). A colleague has tried to compile everything on Ubuntu using clang 14 and that seemed to work well.

The C++ application uses the Rust library to call API functions of our Rust backend, so this is where the network layer (and tokio) is located at. I have also tried to write a Rust program that uses the library to call our server and in that case everything seems to be working fine. Thus I assume that there seems to be an issue the compilation of the C++ program.

I don't really see how I would be able to create a minified example to reproduce this because of the complexity of all related parts, though I am willing to invest more time to provide more details or try out different things to locate the source of the issue. I am hoping that with opening this issue I can get some pointers to what might be going wrong or why this would happen on macOS specifically.

Here is the backtrace:
```
* thread #14, name = 'Thread (pooled)', stop reason = EXC_BAD_ACCESS (code=2, address=0x70000cee2ec8)
    frame #0: 0x000000011a4c7643 libgst_api-3.dylib`__rust_probestack + 47
  * frame #1: 0x000000011818aa1e libgst_api-3.dylib`tokio::sync::mpsc::block::Values$LT$T$GT$::uninitialized::h95648e78733a2006 at block.rs:363
    frame #2: 0x0000000118185550 libgst_api-3.dylib`tokio::sync::mpsc::block::Block$LT$T$GT$::new::h757fcf2389ab0fbd(start_index=0) at block.rs:88:30
    frame #3: 0x0000000117912a9f libgst_api-3.dylib`tokio::sync::mpsc::list::channel::hc95c0583a85ca4dd at list.rs:47:34
    frame #4: 0x0000000117396bda libgst_api-3.dylib`tokio::sync::mpsc::chan::channel::h43a2d9994da93978(semaphore=<unavailable>) at chan.rs:109:20
    frame #5: 0x00000001179e6c23 libgst_api-3.dylib`tokio::sync::mpsc::bounded::channel::hd65d145f6caadc63(buffer=1000000) at bounded.rs:147:20
    frame #6: 0x0000000117f64e59 libgst_api-3.dylib`gst_network::receiving_geometry::make_collection::h35667352ffd38d7e(header=0x0000000115031970) at mod.rs:226:30
    frame #7: 0x0000000117f665f8 libgst_api-3.dylib`gst_network::receiving_geometry::Receiver$LT$M$GT$::get_header::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hf56123ff30cc96f7((null)=ResumeTy @ 0x000070000cf18940) at mod.rs:184:60
    frame #8: 0x0000000116e84dd9 libgst_api-3.dylib`_$LT$core..future..from_generator..GenFuture$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h15279f9b4f066158(self=Pin<&mut core::future::from_generator::GenFuture<gst_network::receiving_geometry::{impl#0}::get_header::{async_fn#0}::{async_block_env#0}<gst_network::schema::LoadFeatureStream, futures_util::stream::try_stream::MapErr<tonic::codec::decode::Streaming<gst_network::schema::LoadFeatureStream>, fn(tonic::status::Status) -> eyre::Report>>>> @ 0x000070000cf19840, cx=0x000070000cf605e0) at mod.rs:91:19
    frame #9: 0x0000000117f65d06 libgst_api-3.dylib`gst_network::receiving_geometry::Receiver$LT$M$GT$::get_header::_$u7b$$u7b$closure$u7d$$u7d$::hce0d542e389aec4b((null)=ResumeTy @ 0x000070000cf249a0) at mod.rs:156:5
    frame #10: 0x0000000116e9f309 libgst_api-3.dylib`_$LT$core..future..from_generator..GenFuture$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h4392bfc37037b590(self=Pin<&mut core::future::from_generator::GenFuture<gst_network::receiving_geometry::{impl#0}::get_header::{async_fn_env#0}<gst_network::schema::LoadFeatureStream, futures_util::stream::try_stream::MapErr<tonic::codec::decode::Streaming<gst_network::schema::LoadFeatureStream>, fn(tonic::status::Status) -> eyre::Report>>>> @ 0x000070000cf25900, cx=0x000070000cf605e0) at mod.rs:91:19
    frame #11: 0x0000000117f68cf6 libgst_api-3.dylib`gst_network::receiving_geometry::Receiver$LT$M$GT$::new::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h57cd236725a80391((null)=ResumeTy @ 0x000070000cf2b050) at mod.rs:153:54
    frame #12: 0x0000000116ea1b89 libgst_api-3.dylib`_$LT$core..future..from_generator..GenFuture$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h48559bd8913abf19(self=Pin<&mut core::future::from_generator::GenFuture<gst_network::receiving_geometry::{impl#0}::new::{async_fn#0}::{async_block_env#0}<gst_network::schema::LoadFeatureStream, futures_util::stream::try_stream::MapErr<tonic::codec::decode::Streaming<gst_network::schema::LoadFeatureStream>, fn(tonic::status::Status) -> eyre::Report>>>> @ 0x000070000cf2bee0, cx=0x000070000cf605e0) at mod.rs:91:19
    frame #13: 0x0000000117f685ad libgst_api-3.dylib`gst_network::receiving_geometry::Receiver$LT$M$GT$::new::_$u7b$$u7b$closure$u7d$$u7d$::h077e87b08b8659bb((null)=ResumeTy @ 0x000070000cf393e0) at mod.rs:147:5
    frame #14: 0x0000000116ec3f59 libgst_api-3.dylib`_$LT$core..future..from_generator..GenFuture$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h87392c97a06ba24c(self=Pin<&mut core::future::from_generator::GenFuture<gst_network::receiving_geometry::{impl#0}::new::{async_fn_env#0}<gst_network::schema::LoadFeatureStream, futures_util::stream::try_stream::MapErr<tonic::codec::decode::Streaming<gst_network::schema::LoadFeatureStream>, fn(tonic::status::Status) -> eyre::Report>>>> @ 0x000070000cf3a340, cx=0x000070000cf605e0) at mod.rs:91:19
    frame #15: 0x0000000116d91bea libgst_api-3.dylib`gst3_api::feature_api::intern::download_feature_with_writer::_$u7b$$u7b$closure$u7d$$u7d$::h908dd93564e3db1e((null)=ResumeTy @ 0x000070000cf4ce78) at intern.rs:626:78
    frame #16: 0x0000000116e85dc9 libgst_api-3.dylib`_$LT$core..future..from_generator..GenFuture$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h16cb009e4101991f(self=Pin<&mut core::future::from_generator::GenFuture<gst3_api::feature_api::intern::download_feature_with_writer::{async_fn_env#0}<gst_writers::FileWriter>>> @ 0x000070000cf4d270, cx=0x000070000cf605e0) at mod.rs:91:19
    frame #17: 0x0000000116d9bef7 libgst_api-3.dylib`gst3_api::feature_api::intern::save_feature_intern::_$u7b$$u7b$closure$u7d$$u7d$::h0d3da8eb59e2afe8((null)=ResumeTy @ 0x000070000cf52f90) at intern.rs:713:5
    frame #18: 0x0000000116eaa1ad libgst_api-3.dylib`_$LT$core..future..from_generator..GenFuture$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h56abacaad33e327c(self=Pin<&mut core::future::from_generator::GenFuture<gst3_api::feature_api::intern::save_feature_intern::{async_fn_env#0}>> @ 0x000070000cf531b0, cx=0x000070000cf605e0) at mod.rs:91:19
    frame #19: 0x0000000117e82662 libgst_api-3.dylib`gst3_api::feature_api::gstr_save_feature::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hf51949e76e2e7d41((null)=ResumeTy @ 0x000070000cf5fc60) at mod.rs:118:17
    frame #20: 0x0000000116ed927d libgst_api-3.dylib`_$LT$core..future..from_generator..GenFuture$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::hadef28c9ba90256b(self=Pin<&mut core::future::from_generator::GenFuture<gst3_api::feature_api::gstr_save_feature::{closure#0}::{async_block_env#0}>> @ 0x000070000cf5feb0, cx=0x000070000cf605e0) at mod.rs:91:19
    frame #21: 0x000000011814c754 libgst_api-3.dylib`_$LT$core..pin..Pin$LT$P$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h72d1dbafa74a0dfd(self=Pin<&mut core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>> @ 0x000070000cf5ff30, cx=0x000070000cf605e0) at future.rs:124:9
    frame #22: 0x000000011761858e libgst_api-3.dylib`_$LT$tracing..instrument..Instrumented$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h0d8a4e8e083dff58(self=Pin<&mut tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>> @ 0x000070000cf5ffb8, cx=0x000070000cf605e0) at instrument.rs:272:9
    frame #23: 0x000000011814b1d2 libgst_api-3.dylib`_$LT$core..pin..Pin$LT$P$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h3823d93979eb74b6(self=Pin<&mut core::pin::Pin<&mut tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>>> @ 0x000070000cf600b8, cx=0x000070000cf605e0) at future.rs:124:9
    frame #24: 0x0000000116848556 libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::CoreGuard::block_on::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hcf105e0f64ddb809 at current_thread.rs:525:48
    frame #25: 0x0000000117f07348 libgst_api-3.dylib`tokio::coop::with_budget::_$u7b$$u7b$closure$u7d$$u7d$::h0ba1152268b69b53(cell=0x000000010a969629) at coop.rs:102:9
    frame #26: 0x000000011790050b libgst_api-3.dylib`std::thread::local::LocalKey$LT$T$GT$::try_with::hd666a57f9d0f55d8(self=0x000000011abd7c90, f={closure_env#0}<core::task::poll::Poll<core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, tokio::runtime::scheduler::current_thread::{impl#9}::block_on::{closure#0}::{closure#0}::{closure_env#0}<core::pin::Pin<&mut tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>>>> @ 0x000070000cf602f0) at local.rs:445:16
    frame #27: 0x00000001178d69c8 libgst_api-3.dylib`std::thread::local::LocalKey$LT$T$GT$::with::h779776e868be570e(self=0x000000011abd7c90, f=<unavailable>) at local.rs:421:9
    frame #28: 0x000000011684603a libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::CoreGuard::block_on::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hdee99f04b0df612c [inlined] tokio::coop::with_budget::h8aa2970714f9822a(budget=(__0 = core::option::Option<> @ 0x000070000cf603a8), f={closure_env#0}<core::pin::Pin<&mut tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>>> @ 0x000070000cf603b0) at coop.rs:95:5
    frame #29: 0x000000011684601b libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::CoreGuard::block_on::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hdee99f04b0df612c [inlined] tokio::coop::budget::h22e372550b163b16(f={closure_env#0}<core::pin::Pin<&mut tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>>> @ 0x000070000cf60370) at coop.rs:72:5
    frame #30: 0x0000000116845fbf libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::CoreGuard::block_on::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hdee99f04b0df612c at current_thread.rs:525:25
    frame #31: 0x00000001167e383a libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::Context::enter::h47abe069df22fc88(self=0x000070000cf609a0, core=0x0000000114f99ff0, f={closure_env#0}<core::pin::Pin<&mut tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>>> @ 0x000070000cf604c0) at current_thread.rs:349:19
    frame #32: 0x000000011681db0d libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::CoreGuard::block_on::_$u7b$$u7b$closure$u7d$$u7d$::h353fab7117ac02df(core=0x0000000114f99ff0, context=0x000070000cf609a0) at current_thread.rs:524:36
    frame #33: 0x0000000116812232 libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::CoreGuard::enter::_$u7b$$u7b$closure$u7d$$u7d$::hceaa1ab1fcc22fb3 at current_thread.rs:595:57
    frame #34: 0x00000001181e28db libgst_api-3.dylib`tokio::macros::scoped_tls::ScopedKey$LT$T$GT$::set::hb3d4d30e988f8535(self=0x000000011acf82d0, t=0x000070000cf609a0, f={closure_env#0}<tokio::runtime::scheduler::current_thread::{impl#9}::block_on::{closure_env#0}<core::pin::Pin<&mut tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>>>, core::option::Option<core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>> @ 0x000070000cf60900) at scoped_tls.rs:61:9
    frame #35: 0x00000001168000df libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::CoreGuard::enter::h0e3565a04fd814e9(self=CoreGuard @ 0x000070000cf609a0, f={closure_env#0}<core::pin::Pin<&mut tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>>> @ 0x000070000cf60938) at current_thread.rs:595:27
    frame #36: 0x00000001168146ac libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::CoreGuard::block_on::h9b9a87ae88177da5(self=<unavailable>, future=Pin<&mut tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>> @ 0x000070000cf609c8) at current_thread.rs:515:19
    frame #37: 0x00000001167bbb40 libgst_api-3.dylib`tokio::runtime::scheduler::current_thread::CurrentThread::block_on::h1909067915e9b05d(self=0x000000011adc36b0, future=<unavailable>) at current_thread.rs:161:24
    frame #38: 0x0000000116f93934 libgst_api-3.dylib`tokio::runtime::Runtime::block_on::heeff1306662bbdc3(self=0x000000011adc36b0, future=Instrumented<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>> @ 0x000070000cf60d68) at mod.rs:490:46
    frame #39: 0x000000011815fda6 libgst_api-3.dylib`gst3_api::common::_$LT$impl$u20$gst3_api..utils..SafeFfiReturnType$LT$T$GT$$u20$for$u20$core..pin..Pin$LT$alloc..boxed..Box$LT$dyn$u20$core..future..future..Future$u2b$Output$u20$$u3d$$u20$core..result..Result$LT$T$C$eyre..Report$GT$$GT$$GT$$GT$::into_result::h857df63361b85d03(self=Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>> @ 0x000070000cf60d18) at common.rs:70:19
    frame #40: 0x000000011746e61d libgst_api-3.dylib`gst3_api::utils::safe_ffi_call::_$u7b$$u7b$closure$u7d$$u7d$::hf3305806e9b27e33 at utils.rs:405:51
    frame #41: 0x0000000116de0835 libgst_api-3.dylib`_$LT$core..panic..unwind_safe..AssertUnwindSafe$LT$F$GT$$u20$as$u20$core..ops..function..FnOnce$LT$$LP$$RP$$GT$$GT$::call_once::h89c5bfc5d2c84fd6(self=AssertUnwindSafe<gst3_api::utils::safe_ffi_call::{closure_env#0}<gst3_api::feature_api::gstr_save_feature::{closure_env#0}, gst3_api::feature_api::types::DownloadResult, gst3_api::utils::ApiResponse<gst3_api::feature_api::types::DownloadResult>, core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>> @ 0x000070000cf60ed0, _args=<unavailable>) at unwind_safe.rs:271:9
    frame #42: 0x000000011751fcce libgst_api-3.dylib`std::panicking::try::do_call::hec2fe8e02740b416(data="P\U00000012\xf6\f") at panicking.rs:464:40
    frame #43: 0x00000001175260bd libgst_api-3.dylib`__rust_try + 29
    frame #44: 0x000000011750a21d libgst_api-3.dylib`std::panicking::try::h0c7b5e96f5e297d8(f=AssertUnwindSafe<gst3_api::utils::safe_ffi_call::{closure_env#0}<gst3_api::feature_api::gstr_save_feature::{closure_env#0}, gst3_api::feature_api::types::DownloadResult, gst3_api::utils::ApiResponse<gst3_api::feature_api::types::DownloadResult>, core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>> @ 0x000070000cf61030) at panicking.rs:428:19
    frame #45: 0x0000000116f4cc6c libgst_api-3.dylib`std::panic::catch_unwind::h5e41ac9e33660fd4(f=AssertUnwindSafe<gst3_api::utils::safe_ffi_call::{closure_env#0}<gst3_api::feature_api::gstr_save_feature::{closure_env#0}, gst3_api::feature_api::types::DownloadResult, gst3_api::utils::ApiResponse<gst3_api::feature_api::types::DownloadResult>, core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output=core::result::Result<gst3_api::feature_api::types::DownloadResult, eyre::Report>>, alloc::alloc::Global>>>> @ 0x000070000cf610a0) at panic.rs:137:14
    frame #46: 0x0000000117461239 libgst_api-3.dylib`gst3_api::utils::safe_ffi_call::h043053d141560620(callback={closure_env#0} @ 0x000070000cf61110) at utils.rs:405:5
    frame #47: 0x00000001175f80b4 libgst_api-3.dylib`gstr_save_feature(client=0x00006000000106d0, download_request=0x000070000cf61790) at mod.rs:96:5
    frame #48: 0x00000001162ccdcc libgst_api-3.dylib`gst3::gstr_save_feature(client=0x00006000000106d0, download_request=0x000070000cf61790) at gst3.h:13054:12
    frame #49: 0x0000000116277a0b libgst_api-3.dylib`gst3::GstClient::gstr_save_feature(this=0x00006000000106d0, download_request=0x000070000cf61790) at gst3.h:14720:16
    frame #50: 0x00000001162775bb libgst_api-3.dylib`GST::ClientUtils::GRPCAccess::SaveFeatureStreamed(this=0x0000600002c68c20, path="XXX", feature=0x000000010b83e3f0, geometryformat=0x000070000cf621d0, srs=GST::Geometry::SRSPtr @ 0x000070000cf61f80, areaLimiter=0x000070000cf62a50, lock=false, commitKey=-1, simplexPropertySelection=const boost::optional<std::__1::vector<long long, std::__1::allocator<long long> > > @ 0x000070000cf62438, write_original_segy_header=false) at GRPCAccess.cpp:8582:30
    frame #51: 0x00000001035e80e7 libgst_gui.dylib`GST::ClientGUI::RequestFeatures::run(this=0x000000010b840460) at AsyncConnection.cpp:631:13
    frame #52: 0x0000000103658580 libgst_gui.dylib`GST::ClientGUI::AsyncConnectionRunner::AsyncConnectionRunner(this=0x000000010b83e7f8)::$_2::operator()() const at AsyncConnection.cpp:3175:28
    frame #53: 0x000000010365835d libgst_gui.dylib`boost::detail::function::void_function_obj_invoker0<GST::ClientGUI::AsyncConnectionRunner::AsyncConnectionRunner(GST::ClientGUI::TaskManager&, GST::ClientGUI::AsyncConnection*)::$_2, void>::invoke(function_obj_ptr=0x000000010b83e7f8) at function_template.hpp:158:11
    frame #54: 0x0000000103738034 libgst_gui.dylib`boost::function0<void>::operator(this=0x000000010b83e7f0)() const at function_template.hpp:763:14
    frame #55: 0x000000010373793f libgst_gui.dylib`GST::ClientGUI::Task::run(this=0x000000010b83e7a0) at Task.cpp:129:4
    frame #56: 0x0000000106364743 QtCore`QThreadPoolThread::run() + 117
    frame #57: 0x0000000106361fb0 QtCore`QThreadPrivate::start(void*) + 322
    frame #58: 0x000000010030ac0d libsystem_pthread.dylib`_pthread_start + 125
    frame #59: 0x0000000100312ccf libsystem_pthread.dylib`thread_start + 15
```

Please let me know if I can provide more details! :)

**Comment by Darksonn** [comment]
Reading the documentation for `__rust_probestack` makes this sound like a stack overflow. How large is the type used as the message type in the mpsc channel?

**Comment by MrMuetze** [comment]
Hey, thanks for the quick reply. Your suggestion was invaluable!

We checked what would happen if we `Box` the messages related to the issue and we were able to circumvent the stack overflow. The message itself only seems to be 792 bytes large. We suspect that maybe we put too many other things on the stack and even a small increase for this particular message would then lead to the panic I have experienced. This might also explain why the issues does vanish if we build everything in release mode (because memory might be used more efficiently then).

All in all this seems to be something that we have to handle on our side and shouldn't concern `tokio-rs`, so I close this issue for now.

Thanks again!
